worldgen/scripts/terrain_gen/carvers/floating_island_carver.py

```
@@ -0,0 +1,197 @@
+        self.drip_scale = max(8.0, drip_scale)
+        self.drip_sharpness = max(0.25, drip_sharpness)
+
+    def _silhouette(self, wx: int, wz: int, seed: int) -> float:
+        # Warped 2.5D-ish blob sampled at the island altitude → island plan view.
+        raw = warped_fbm_3d(
+            float(wx), float(self.altitude), float(wz),
+            scale=self.scale, warp_scale=180.0, warp_strength=40.0, seed=seed + 5,
+        )
+        return float((raw + 1.0) * 0.5)
+
+    def _gate_mask(self, wx, surface_y, wz, seed, active):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `_gate_mask`

(ANN202)